### PR TITLE
Add functionality to dynamically add research interest

### PR DESCRIPTION
**Pull request by linhpha**
This PR allows students to add research interest dynamically and delete undergrad university from their profile. It also requires students to fills in all the fields when adding their education to the profile page.
## Issue related

#246

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Front end change

## Code coverage
98.14%

## Screenshot (if any change in front end)
image

image

### CODE REVIEW

File: app/assets/javascripts/edit_profile.js [hunk 1]
```diff
@@ -191,4 +191,15 @@ $(document).ready(function () {
...
     });
-});
\ No newline at end of file
+});
+
+
+var add_new_research_interest = function() {
+
+    $('#add_new_research_interest').click(function() {
+        console.log('here');
```
Comment (hpitawela) [inline app/assets/javascripts/edit_profile.js]: Better remove this debug line.

File: app/controllers/profiles_controller.rb [hunk 2]
```diff
@@ -425,199 +452,200 @@ def filter
...
-      undergrad_uni = UndergradUniversity.find_by_id(params[:undergrad_university])
-      profiles_other = undergrad_uni.profiles.where("profiles.gre_quant >= #{greq_low} AND profiles.gre_quant <= #{greq_high} AND "+
-                                                        "profiles.gre_verbal >= #{grev_low} AND profiles.gre_verbal <= #{grev_high} AND "+
-                                                        "profiles.degree_objective_phd >= #{phdo_low} AND profiles.degree_objective_phd <= #{phdo_high} AND "+
-                                                        "profiles.degree_objective_master >= #{msob_low} AND profiles.degree_objective_master <= #{msob_high}").all
-      unless params[:research_interests].blank?
-        if params[:research_interests].to_s =~ /^any$/
-          profiles = profiles_other
+        undergrad_uni = UndergradUniversity.find_by_id(params[:undergrad_university])
+        profiles_other = undergrad_uni.profiles.where("profiles.gre_quant >= #{greq_low} AND profiles.gre_quant <= #{greq_high} AND "+
```
Comment (Avanti0532) [inline app/controllers/profiles_controller.rb]: Do you have changes in filter function of profiles controller? If not, can you pull from master again to make sure that your PR doesn't show changes in filter function?

File: app/assets/javascripts/edit_profile.js (hunk 1 above)
Comment (linhpha) [reply app/assets/javascripts/edit_profile.js]: :+1:

File: app/controllers/profiles_controller.rb (hunk 2 above)
Comment (linhpha) [reply app/controllers/profiles_controller.rb]: I already pulled from master but I don't know why it showed changes in my PR. I might have re-indent the file altogether

Comment (hpitawela) [reply app/controllers/profiles_controller.rb]: I would suggest copy filter method from master in remote repo and paste in your file.

Comment (linhpha) [reply app/controllers/profiles_controller.rb]: I copied the latest filter method from master

**Pull request merged**